test/units/TEST-72-SYSUPDATE.sh

```diff
@@ -227,15 +239,26 @@ EOF
     test ! -f "$WORKDIR/xbootldr/EFI/Linux/uki_v1.efi.extra.d/extra.addon.efi"
     test ! -d "$WORKDIR/xbootldr/EFI/Linux/uki_v1.efi.extra.d"
 
-    # Create fourth version, and update through a file:// URL. This should be
+    # Create fourth version, update using updatectl and verify it replaced the
+    # second version
+    new_version "$sector_size" v4
+    systemctl start systemd-sysupdated
+    updatectl check
+    updatectl update
+    # User-facing updatectl returns 0 if there's no updates, so use the low-level
+    # utility to make sure we did upgrade
+    (! "$SYSUPDATE" --verify=no check-new )
+    verify_version "$blockdev" "$sector_size" v4 2 4
+
```

Comment:
I'd add another section here, that just runs the various `updatectl` commands, just to test for obvious crashes or memory leaks via the sanitizers. For example, something along these lines:

```
# Next, let's run updatectl's various inspection commands. We're not testing for specific output,
# but this will at least catch obvious crashes and allow updatectl to run under the various sanitizers. We create a component so that updatectl has multiple targets to list.
install -Dm644 "$CONFIGDIR/01-first.conf" /run/sysupdate.test.d/01-first.conf
updatectl list
updatectl list host
updatectl list host@v4
updatectl vacuum host
rm -r /run/sysupdate.test.d
```

Reply:
In this section, no need to test `check` (should be tested along with `update`), `update` (already tested above), and `vacuum` (I'll test it as part of my optional-features work)